doc(pe): Document policy options (#601)
armory/docs: content/en/docs/plugin-guide/plugin-policy-engine/packages/spinnaker.execution/stages.before/jenkins.md

---
title: "spinnaker.execution.stages.before.jenkins"
linktitle: "jenkins"

> description: "A policy targeting this object is run before executing each task in a jenkins stage."

Reviewer: ```suggestion
description: "A policy targeting this object runs before executing each task in a Jenkins stage."
```

          "loopStart": false,
          "name": "getBuildProperties",
          "stageEnd": false,
          "stageStart": false,
          "startTime": null,
          "status": "NOT_STARTED"
        },
        {
          "endTime": null,
          "id": "5",
          "implementingClass": "com.netflix.spinnaker.orca.igor.tasks.GetBuildArtifactsTask",
          "loopEnd": false,
          "loopStart": false,
          "name": "getBuildArtifacts",
          "stageEnd": true,
          "stageStart": false,
          "startTime": null,
          "status": "NOT_STARTED"
        }
      ],
      "type": "jenkins"
    },
    "user": {
      "isAdmin": false,
      "roles": [],
      "username": "myUserName"
    }
  }
}
```
</details>

## Keys

### input.pipeline

| Key                                               | Type      | Description                                                                                                                                       |
| ------------------------------------------------- | --------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `input.pipeline.application`                      | `string`  | The name of the Spinnaker application to which this pipeline belongs.                                                                             |

Reviewer: ```suggestion
| `input.pipeline.application`                      | `string`  | The name of the application to which this pipeline belongs.                                                                             |
```

| `input.pipeline.authentication.allowedAccounts[]` | `string`  | The list of accounts to which the user this stage is running as has access.                                                                       |

> | `input.pipeline.authentication.user`              | `string`  | The Spinnaker user initiating the change.                                                                                                         |

Reviewer: ```suggestion
| `input.pipeline.authentication.user`              | `string`  | The user initiating the change.                                                                                                         |
```
